rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -916,6 +916,13 @@ public List<ResolvedDependency> resolveDependencies(Scope scope, Map<GroupArtifa
                         } else if (contains(dependencies, ga, d.getClassifier())) {
                             // we've already resolved this previously and the requirement didn't change,
                             // so just skip and continue on
+                            ResolvedDependency includedBy = dd.getDependent();
+                            if (includedBy != null) {
+                                if (includedBy.getOmitDependencies().isEmpty()) {
+                                    includedBy.unsafeSetOmitDependencies(new ArrayList<>());
+                                }
+                                includedBy.getOmitDependencies().add(new GroupArtifactVersion(d.getGroupId(), d.getArtifactId(), d.getVersion()));
```

Comment:
Also here the generated getter might return null; as an alternative we could create an explicit getter that always returns a collection.

Reply:
`getOmitDependencies()` never return null now.

Reply:
`getOmitDependencies()` never return null now